packages/m365_defender/data_stream/event/elasticsearch/ingest_pipeline/pipeline_app_and_identity.yml

```
@@ -1,468 +1,559 @@
 ---
 description: Pipeline for processing Apps & Identities Event logs.
 processors:
+  # Event categorization
   - set:
       field: event.kind
-      value: [event]
-  - set:
+      value: event
+      tag: set_event_kind
+  - append:
       field: event.category
-      value: [authentication]
-      if: ctx.m365_defender?.event?.category?.toLowerCase().contains('identitylogonevents')
-  - set:
-      field: event.type
-      value: [info]
-  - rename:
-      field: json.properties.TargetDeviceName
-      target_field: m365_defender.event.target.device_name
-      ignore_missing: true
-  - set:
-      field: destination.domain
-      copy_from: m365_defender.event.target.device_name
-      ignore_empty_value: true
+      value: authentication
+      tag: append_event_category
+      if: ctx.m365_defender?.event?.category != null && ctx.m365_defender?.event?.category?.toLowerCase().contains('identitylogonevents')
```

```suggestion
      if: ctx.m365_defender?.event?.category != null && ctx.m365_defender.event.category.toLowerCase().contains('identitylogonevents')
```
Though comment above. It may be worth adding a comment about this invariant so that future people don't get upset.